operator/pkg/reconciliation/reconcile_services.go

```diff
@@ -90,19 +90,23 @@ func (rc *ReconciliationContext) CheckHeadlessServices() result.ReconcileResult
 			return result.Error(err)
 
 		} else {
-			// if we found the service already, check if the labels are right
-			currentLabels := currentService.GetLabels()
-			desiredLabels := desiredSvc.GetLabels()
-			shouldUpdateLabels := !reflect.DeepEqual(currentLabels, desiredLabels)
-			if shouldUpdateLabels {
-				logger.Info("Updating labels",
+			// if we found the service already, check if they need updating
+			if !resourcesHaveSameHash(currentService, desiredSvc) {
```

Comment:
It turns out this change isn't necessary because we removed `managed-by` from the `selector` of our services. However, it's still a good idea so here it is.